docker-compose.yaml

```diff
@@ -1,6 +1,30 @@
 version: "3.7"
 
 services:
+  
+  traefik:
+    image: "traefik:v3.1"
+    command:
+      - "--api.insecure=true"
+      - "--providers.docker=true"
+      - "--providers.docker.exposedbydefault=false"
+      - "--providers.docker.network=default"
+      - "--entrypoints.web.address=:80"
+      - "--api"
+    volumes:
+      - "/var/run/docker.sock:/var/run/docker.sock:ro"
+    ports:
+      - 80:80
+      # management console
+      - 8080:8080
+    labels:
+      - "traefik.enable=true"
+      - "traefik.http.routers.http-catchall.rule=hostregexp(`{host:.+}`)"
+      - "traefik.http.routers.http-catchall.entrypoints=web"
+      # auth middleware with user / password
+      # password: test username: test
+      - "traefik.http.middlewares.my-auth.basicauth.users=test:$$apr1$$H6uskkkW$$IgXLP6ewTrSuBkTrqE8wj/,test2:$$apr1$$d9hr9HBB$$4HxwgUir3HP4EsggP/QNo0"
```

Comment:
Change this in prod, just here for an example hashed passoword. test test works.